gcp/app-engine/main.py

```diff
@@ -1,147 +1,65 @@
+        return jsonify(schema.dump(data)), 200
+    except Exception as e:
+        print(f'Error querying get_pga_tournaments tournament_id={tournament_id}: {e}', e)
+        return json.dumps({"error": "Internal server error"}), 500
+    finally:
+        db.close()
 
 
 @app.route(PUBLIC_API_PREFIX + '/pga-leaderboard-players', methods=['GET'])
 @validate_token
 def get_pga_leaderboard_players(user_email):
     print(f"Got get_pga_leaderboard_players request from user: {user_email}")
-    query = "SELECT data FROM golfbetter.pga_leaderboard_players where 1=1"
+    id = request.args.get('id', None)
     tournament_id = request.args.get('tournamentId', None)
-    if tournament_id is not None:
-        query = query + " and tournament_id = '" + str(tournament_id) + "'"
     player_id = request.args.get('id', None)
-    if player_id is not None:
-        query = query + " and id = '" + str(player_id) + "'"
-    pool = get_db_connection()
-    with pool.connect() as db_conn:
-        records = db_conn.execute(sqlalchemy.text(query)).fetchall()
-        return json.dumps([row[0] for row in records], default=json_serial), 200
+    db = SessionLocal()
+    try:
+        data = [t.data for t in get_leaderboard_players(db, id, player_id, tournament_id)]
+        schema = PlayerRowV3Schema(many=True)
+        return jsonify(schema.dump(data)), 200
+    except Exception as e:
+        print(f'Error querying get_pga_leaderboard_players id={id} tournament_id={tournament_id} player_id={player_id}: {e}', e)
+        return json.dumps({"error": "Internal server error"}), 500
+    finally:
+        db.close()
 
 @app.route(PUBLIC_API_PREFIX + '/pga-player-scorecards', methods=['GET'])
 @validate_token
 def get_pga_player_scorecards(user_email):
     print(f"Got get_pga_player_scorecards request from user: {user_email}")
-    query = "SELECT data FROM golfbetter.pga_player_scorecards where 1=1"
+    id = request.args.get('id', None)
     tournament_id = request.args.get('tournamentId', None)
-    if tournament_id is not None:
-        query = query + " and tournament_id = '" + str(tournament_id) + "'"
     player_id = request.args.get('id', None)
-    if player_id is not None:
-        query = query + " and id = '" + str(player_id) + "'"
-    pool = get_db_connection()
-    with pool.connect() as db_conn:
-        records = db_conn.execute(sqlalchemy.text(query)).fetchall()
-        return json.dumps([row[0] for row in records], default=json_serial), 200
+    db = SessionLocal()
+    try:
+        data = [t.data for t in get_player_scorecards(db, id, player_id, tournament_id)]
+        schema = ScorecardV3Schema(many=True)
+        return jsonify(schema.dump(data)), 200
+    except Exception as e:
+        print(f'Error querying get_pga_player_scorecards id={id} tournament_id={tournament_id} player_id={player_id}: {e}', e)
+        return json.dumps({"error": "Internal server error"}), 500
+    finally:
+        db.close()
```

Comment:
will this fail if the exception caught above is a database connection error? not a big deal though